backend/src/models/entry/entryAnalysis.ts

```diff
@@ -85,64 +73,36 @@ async function removeLegacyApiKey(configId: string): Promise<string> {
       }
     }
   }
-  return config.model.analysis;
-}
 
-async function getAnalysisCompletion(
-  configModelAnalysis: string,
-  content: string
-): Promise<any> {
-  if (process.env.OPENAI_API_KEY === undefined) {
-    throw new Error('OPENAI_API_KEY undefined. Must be set to get analysis');
+  if (!process.env.OPENAI_API_KEY) {
+    throw new Error('API key is not available)');
   }
-  const cdGpt = new CdGpt(
-    process.env.OPENAI_API_KEY,
-    configModelAnalysis,
-    '',
-    0.7
-  );
+
+  const cdGpt = new CdGpt(process.env.OPENAI_API_KEY, config.model.analysis);
 
   cdGpt.seedAnalysisMessages();
   cdGpt.addUserMessage({ analysis: content });
 
-  const analysisCompletion: any = await cdGpt.getAnalysisCompletion();
+  const analysisCompletion = await cdGpt.getAnalysisCompletion();
```

Comment:
I left the type as any here because I thought it would make more sense to do the type changes in CdGpt in a separate issue rather than do it together with the models migration. We can do it here, but it's going to make a big PR even bigger. In the future, I think we should do things like this in a separate issue.

Reply:
We definitely want to mention that in a TODO dev comment with an associated Issue. I wouldn't have known the intention without it. But yes, next time we can add inline comments to clarify intentions.